### ISSUE TITLE
Custom Tab with Custom Content on "Edit Pod" Pages

### ISSUE THREAD

**Issue by tristanleboss** [pods-framework/pods#2849]
Hello,

I was planning to develop a "Manage Metaboxes" tab (to create "Metaboxes") for a customer.

I started digging in the code and found it was possible to add a tab in "pods/classes/PodsAdmin.php" but, unfortunately, the tab rendering code in "pods/ui/admin/setup-edit.php" only renders the fields defined in $tab_options: there is no way to hook there to render a totally custom content.

I think that would give people more freedom in order to extend pods.

Tristan

PS: I don't know how to label this issue as a feature request.

**Comment by sc0ttkclark** [comment]
Love where you're going with this request, this is exactly what I had in mind for all the related work I've been putting into Pods 3.0 for this. It's got the filters and actions that you're looking for. The downside is 3.0 isn't ready for development or production use, we're still merging in Pods 2.5.x changes into it and making sure it passes all of our unit tests. Some parts of the admin are broken as we're in the middle of implementing the new loop fields.

I'll see what we can do for you, I've added some alternative screens before and it should be doable here.

**Comment by tristanleboss** [comment]
The lack of metabox support can't last longer: we have to do something.

I found a hacky way to do it:
- Create a new field type which will host the content I want to show on this new tab
- Create the tab with one instance of the newly created field type

This way, I have a display and a save function.

I created a "reset" css which will give me the whole space:
`<style type="text/css">
	.pods-nav-tab#pods-metaboxes .pods-field-option {
		border-bottom: none;
		background-color: none;
		padding: 0;
	}
	
	.pods-nav-tab#pods-metaboxes .pods-field-option > label {
		display: none;
	}
	
	.pods-nav-tab#pods-metaboxes .pods-field-option > .pods-field {
		margin: 0;
	}
	
	.pods-nav-tab#pods-metaboxes .pods-field-option > .pods-field:hover {
		background: none;
	}
</style>`

Problem, I need to keep track of field change (new field added, field deleted, field name modified) made in "Manage Fields" tab. You probably know your plugin better than me: do you have any instant ideas on what would be the easiest way to achieve it? I'm open to a solution as crazy as mine.

**Comment by sc0ttkclark** [comment]
We have meta box support via code -- pods_group_add() will do it, but
you're looking to register them through the UI, not just code, right?

>

-- 


Best Regards,

Scott Kingsley Clark
Developer

------------------------------------
E-mail: lol@scottkclark.com
Website: http://scottkclark.com/
Phone: 214.233.4048

Google: sc0ttkclark@gmail.com
Skype: sc0ttkclark
Slack: sc0ttkclark on wordpress + podswp
------------------------------------

**Comment by tristanleboss** [comment]
I already know about the API: I will use it to create the metaboxes. But, I indeed want to do a simple UI to manage this. 

My new "Manage Metaboxes" tab will be something like "Tabify Edit Screen". Instead of creating "Tabs", you will create "Metaboxes". And, instead of putting "Metaboxes" into "Tabs", you will put "Fields' into "Metaboxes".

I plan to save everything in a new table `wp_podsmetaboxes` with fields: pod_name, metabox_name metabox_title, metabox_fields (comma separated list of field names).

The problem comes from the fact that the "Edit Pod" page is rendered all at once. So, if someone apply some changes to the fields on the "Manage Fields" type, I need to be able to LIVE UPDATE my "Manage Metaboxes" tab.

**Comment by sc0ttkclark** [comment]
I've got some ideas for you, will let you know when I have some time to think it over this weekend.

**Comment by tristanleboss** [comment]
Ok, I finished my plugin :+1: I know have a Metaboxes manager inside the "Edit Pod". I only used jQuery for the UI.

I encountered a few glitches:

- Lack of documentation/examples for Pods. For example, I struggled to find a way to quickly get the value of a pods' option. I ended up using PodsAPI::load_pod and accessing the 'options' in the resulting array.

- I created a new field type to encapsulate my UI which I then used it as an option on a new tab. But it seems that most the magic methods (pre_save/...) of the extended PodsField class are not used when a field type is used as an option?

- The magic method "PodsField::data" is called with its 4th arguement $pod = null when the field type is used as an option.

- I don't know if it's a Pods or WordPress problem but if multiple metabox have the same label, only the first one is registered.
